src/include/processor/operator/persistent/rel_batch_insert.h

```diff
@@ -29,6 +29,14 @@ struct RelBatchInsertPrintInfo final : OPPrintInfo {
         : OPPrintInfo(other), tableName(other.tableName) {}
 };
 
+struct RelBatchInsertProgressSharedState {
+    std::mutex mtx;
```

Comment:
If you make the change suggested on `PartitionerSharedState::getNextPartition`, you don't need a mutex here.